lib/pbench/agent/tool_meister_start.py

```diff
@@ -0,0 +1,1296 @@
+            else:
+                logger.warning("error %s (%s)", e.message, cause)
+            status = e.status
+        else:
+            status = ReturnCode.INITFAILED
+            logger.exception("Unexpected exception in outer try")
+        recovery.cleanup()
+        return status
+
+
+_NAME_ = "pbench-tool-meister-start"
+
+
+def main():
+    parser = ArgumentParser(f"{_NAME_} [--sysinfo <list of system information items>]")
+    parser.add_argument(
+        "--sysinfo",
+        dest="sysinfo",
+        default=None,
+        help="The list of system information items to be collected.",
+    )
+    parser.add_argument(
+        "--orchestrate",
+        dest="orchestrate",
+        default=os.environ.get("PBENCH_ORCHESTRATE", _def_orchestrate_choice),
+        choices=_orchestrate_choices,
+        help=(
+            "The `create` keyword directs the command to create the various"
+            " instances of the Redis server, Tool Data Sink, and Tool"
+            " Meisters, while the `existing` keyword directs the command to"
+            " use existing instances of all three. The default is `create`."
+        ),
+    )
+    parser.add_argument(
+        "--redis-server",
+        dest="redis_server",
+        default=os.environ.get("PBENCH_REDIS_SERVER", None),
+        help=(
+            "Specifies the IP/port to use for the Redis server - if not"
+            " present, the defaults are used, ${_pbench_full_hostname}:"
+            f"{def_redis_port};"
+            " the specified value can take either of two forms:"
+            " `<bind host>:<port>;<host>:<port>`, a semi-colon separated"
+            " IP/port specified for both how the Redis server will bind"
+            " itself, and how clients will connect; `<host>:<port>`, the"
+            " IP/port combination is used both for binding and connecting"
+            " (NOTE: binding is not used with --orchestrate=existing);"
+        ),
+    )
+    parser.add_argument(
+        "--tool-data-sink",
+        dest="tool_data_sink",
+        default=os.environ.get("PBENCH_TOOL_DATA_SINK", None),
+        help=(
+            "Specifies the IP/port to use for the Tool Data Sink - if not"
+            " present, the defaults are used, ${_pbench_full_hostname}:"
+            f"{def_wsgi_port};"
+            " the specified value can take either of two forms:"
+            " `<bind host>:<port>;<host>:<port>`, a semi-colon separated"
+            " IP/port specified for both how the Tool Data Sink will bind"
+            " itself, and how clients will connect; `<host>:<port>`, the"
+            " IP/port combination is used both for binding and connecting"
+            " (NOTE: binding is not used with --orchestrate=existing);"
+        ),
+    )
+    parser.add_argument(
+        "tool_group",
+        help="The tool group name of tools to be run by the Tool Meisters.",
+    )
+    parsed = parser.parse_args()
+    status = start(sys.argv[0], parsed)
+    sys.exit(status)
```

Comment:
This all (except for the call to `start()`, I suppose) seems like it should stay in the CLI script -- the entry to this module should have an API-style interface.

Reply:
While I stopped short of converting these utility CLIs to click, the current division of labor is very click-like. I mean to refactor the real logic into a reusable class hierarchy, which will make the division between `main` and the library logic more evident.